### PR TITLE
2180 apiv3 dataset

### PR DESCRIPTION
**Pull request by aviveiros11**
#### Any background context you want to provide?
The effort to move to API v3 - specifical datasets
#### What's this PR do?
translates API v2 into v3 with auto schema helper 
#### How should this be manually tested?
Use the `Try It` functionality rendered by the API page on SEED for v3
#### What are the relevant tickets?
#2180

### CODE REVIEW

**Comment by aviveiros11** [comment]
@adrian-lara, @axelstudios this is just about there.  I wanted to get some more clarification on how we are defining "deactivate" in task 3 of #2180.  If we are just refactoring to deactivate I believe we have to touch the Model? Or are we looking to refactor the delete functionality to keep the dataset but flag it as deactivated?

**Comment by axelstudios** [comment]
@aviveiros11 I'm pretty sure that means don't physically delete the row when the DELETE method is called, but instead toggle the `deleted` boolean column to true:

image

**Comment by adrian-lara** [comment]
> @adrian-lara, @axelstudios this is just about there. I wanted to get some more clarification on how we are defining "deactivate" in task 3 of #2180. If we are just refactoring to deactivate I believe we have to touch the Model? Or are we looking to refactor the delete functionality to keep the dataset but flag it as deactivated?

Definitely +1 to what Alex said. It might be the case that this is already happening ("deactivate" flagging is happening instead of actual DB row deletions), so definitely double check that.

As far as potentially making changes, I think you're asking if the "deactivate" logic requires changing logic at the Model level or the ViewSet level? If I'm understanding that question correctly, I guess we'd have to answer the question of "Do we want this to happen only when this endpoint is hit or whenever anything (other models, utils, etc) tries to delete a record?".

**Comment by aviveiros11** [comment]
Confirmed with Alex that the column in fact was being updated.  Ready for review.

File: seed/views/v3/datasets.py [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+
+        return JsonResponse({
+            'status': 'success',
+            'dataset': dataset,
+        })
+
+    @api_endpoint_class
+    @ajax_request_class
+    @has_perm_class('requires_member')
+    def destroy(self, request, pk=None):
```
Comment (adrian-lara) [inline seed/views/v3/datasets.py:257]: Deactivates as expected 👍 
Screen Shot 2020-05-06 at 4 31 16 PM

File: seed/utils/api_schema.py [hunk 2]
```diff
@@ -11,7 +11,8 @@ class AutoSchemaHelper(SwaggerAutoSchema):
         'interger_list': openapi.Schema(
             type=openapi.TYPE_ARRAY,
             items=openapi.Schema(type=openapi.TYPE_INTEGER)
-        )
+        ),
+        'string': openapi.Schema(type=openapi.TYPE_STRING)
```
Comment (adrian-lara) [inline seed/utils/api_schema.py:15]: 👍

File: seed/views/v3/datasets.py [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+from seed.models import obj_to_dict
+from seed.utils.api import api_endpoint_class
+from seed.utils.time import convert_to_js_timestamp
+
+from seed.utils.api_schema import AutoSchemaHelper
+
+_log = logging.getLogger(__name__)
+
+
+class DatasetSchema(AutoSchemaHelper):
```
Comment (adrian-lara) [inline seed/views/v3/datasets.py:28]: Let me know if there's anything notable changed outside of this class, the AutoSchemaHelper import, and the application of this DatasetSchema class to the -ViewSet (`swagger_schema=...`). 

I assumed these were the only "real" changes this file and mostly everything else was just copied over.

Comment (aviveiros11) [reply seed/views/v3/datasets.py:28]: Yes exactly, just wanted to follow the AutoSchmaHelper pattern but mostly everything was just copied over 👍 .

**Pull request merged**